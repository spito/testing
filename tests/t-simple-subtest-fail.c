#include <cut.h>
#include <stdlib.h>

TEST(withSubtest) {
    int i = 1;
    SUBTEST("S1") {
        CHECK(0);
        ASSERT(i == 1);
    }
    ++i;
    SUBTEST("S2 (failing)") {
        ASSERT(0);
    }
    ++i;
    SUBTEST("S3") {
        ASSERT(i == 3);
    }
    ++i;
    SUBTEST("S4") {
        int *n = NULL;
        *n = 4;
    }
    ++i;
    ASSERT(i == 5);
}
