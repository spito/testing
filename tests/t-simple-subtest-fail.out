[  1] withSubtest: 4 subtests
    S1.....................................................................FAIL
        check '0' (simple-subtest-fail.c:7)

    S2 (failing)...........................................................FAIL
        assert '0' (simple-subtest-fail.c:12)

    S3.......................................................................OK
    S4................................................................SIGNALLED
        signal: SIGSEGV (11)

[  1] withSubtest (overall)................................................FAIL

Summary:
  tests:          1
  succeeded:      0
  filtered out:   0
  suppressed:     0
  skipped:        0
  failed:         1
